src/api/hooks/useGetANS.ts

```diff
@@ -2,25 +2,40 @@ import {useEffect, useState} from "react";
 import {NetworkName} from "../../constants";
 import {useGlobalState} from "../../GlobalState";
 
-function getFetchNameUrl(network: NetworkName, address: string) {
+function getFetchNameUrl(
+  network: NetworkName,
+  address: string,
+  isPrimary: boolean,
+) {
   if (network !== "testnet" && network !== "mainnet") {
     return undefined;
   }
 
-  return `https://www.aptosnames.com/api/${network}/v1/name/${address}`;
+  return isPrimary
+    ? `https://www.aptosnames.com/api/${network}/v1/primary-name/${address}`
+    : `https://www.aptosnames.com/api/${network}/v1/name/${address}`;
 }
 
 export function useGetNameFromAddress(address: string) {
   const [state, _] = useGlobalState();
   const [name, setName] = useState<string | undefined>();
-  const url = getFetchNameUrl(state.network_name, address);
 
   useEffect(() => {
-    if (url !== undefined) {
+    const primaryNameUrl = getFetchNameUrl(state.network_name, address, true);
+    if (primaryNameUrl !== undefined) {
       const fetchData = async () => {
-        const response = await fetch(url);
-        const {name} = await response.json();
-        setName(name);
+        const primaryNameResponse = await fetch(primaryNameUrl);
+        const {name: primaryName} = await primaryNameResponse.json();
+
+        if (primaryName) {
+          setName(primaryName);
```

Comment:
as discussed offline, primaryName has `.apt`, which we will want to remove when displaying the name on explorer.

Reply:
will change it in the ans repo